packages/beacon-node/src/chain/archiver/archiveBlocks.ts

```diff
@@ -81,19 +82,25 @@ export async function archiveBlocks(
   }
 
   // Delete expired blobs
-  // Keep only `[max(GENESIS_EPOCH, current_epoch - MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS), current_epoch]`
+  // Keep only `[current_epoch - max(MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS, archiveBlobEpochs)]
+  // if archiveBlobEpochs set to Infinity do not prune`
   if (finalizedPostDeneb) {
-    const blobSidecarsMinEpoch = currentEpoch - config.MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS;
-    if (blobSidecarsMinEpoch >= config.DENEB_FORK_EPOCH) {
-      const slotsToDelete = await db.blobSidecarsArchive.keys({lt: computeStartSlotAtEpoch(blobSidecarsMinEpoch)});
-      if (slotsToDelete.length > 0) {
-        await db.blobSidecarsArchive.batchDelete(slotsToDelete);
-        logger.verbose(
-          `blobSidecars prune: batchDelete range ${slotsToDelete[0]}..${slotsToDelete[slotsToDelete.length - 1]}`
-        );
-      } else {
-        logger.verbose(`blobSidecars prune: no entries before epoch ${blobSidecarsMinEpoch}`);
+    if (archiveBlobEpochs !== Infinity) {
+      const blobsArchiveWindow = Math.max(config.MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS, archiveBlobEpochs ?? 0);
+      const blobSidecarsMinEpoch = currentEpoch - blobsArchiveWindow;
```

Comment:
I guess we are fine with this being a negative number, in that case, do we even need to handle `Infinity` explicitly?

Reply:
> although the log output would be funny

It is not that bad

```
blobSidecars prune: no entries before epoch -Infinity
```

but I guess the other log is more explicit about it

Reply:
its good to be explicit not just for code correctness but also because for code readability and understanding which is one of the lodestar's objective (to stay readable and declarative)